web/src/components/UploadAirgapBundle.tsx

```diff
@@ -0,0 +1,738 @@
+import React, { useEffect, useReducer, useRef } from "react";
+import { useHistory, useRouteMatch } from "react-router";
+import classNames from "classnames";
+import { KotsPageTitle } from "@components/Head";
+import isEmpty from "lodash/isEmpty";
+import Modal from "react-modal";
+import CodeSnippet from "@src/components/shared/CodeSnippet";
+import MountAware from "@src/components/shared/MountAware";
+import AirgapUploadProgress from "@features/Dashboard/components/AirgapUploadProgress";
+import LicenseUploadProgress from "./LicenseUploadProgress";
+import AirgapRegistrySettings from "./shared/AirgapRegistrySettings";
+import { Utilities } from "../utilities/utilities";
+import { AirgapUploader } from "../utilities/airgapUploader";
+
+import "../scss/components/troubleshoot/UploadSupportBundleModal.scss";
+import "../scss/components/Login.scss";
+
+const COMMON_ERRORS = {
+  "HTTP 401": "Registry credentials are invalid",
+  "invalid username/password": "Registry credentials are invalid",
+  "no such host": "No such host",
+};
+
+import { KotsParams } from "@types";
+
+type Props = {
+  appName: string | null;
+  appsListLength: number;
+  logo: string | null;
+  fetchingMetadata: boolean;
+  onUploadSuccess: () => Promise<void>;
+  showRegistry: boolean;
+};
+
+type RegistryDetails = {
+  hostname: string;
+  isReadOnly: boolean;
+  namespace: string;
+  password: string;
+  username: string;
+};
+
+type ResumeResult = {
+  error?: string;
+  hasPreflight: boolean;
+  isConfigurable: boolean;
+};
+
+type State = {
+  airgapUploader: AirgapUploader | null;
+  bundleFile: {
+    name: string;
+  } | null;
+  displayErrorModal?: boolean;
+  errorMessage: string;
+  fileUploading: boolean;
+  registryDetails: RegistryDetails | null;
+  preparingOnlineInstall: boolean;
+  supportBundleCommand?: string | string[];
+  showSupportBundleCommand: boolean;
+  simultaneousUploads?: number;
+  uploadProgress: number;
+  uploadSize: number;
+  uploadResuming: boolean;
+  viewOnlineInstallErrorMessage: boolean;
+};
+const UploadAirgapBundle = (props: Props) => {
+  const [state, setState] = useReducer(
+    (currentState: State, newState: Partial<State>) => ({
+      ...currentState,
+      ...newState,
+    }),
+    {
+      airgapUploader: null,
+      bundleFile: null,
+      errorMessage: "",
+      fileUploading: false,
+      registryDetails: null,
+      preparingOnlineInstall: false,
+      showSupportBundleCommand: false,
+      uploadProgress: 0,
+      uploadSize: 0,
+      uploadResuming: false,
+      viewOnlineInstallErrorMessage: false,
+    }
+  );
+
+  const emptyHostnameErrMessage = 'Please enter a value for "Hostname" field';
+  const match = useRouteMatch<KotsParams>();
+  const history = useHistory();
+  const appSlug = match.params.slug;
+  // TODO: refactor the /resume fetching so this isn't necessary
+  const onlineInstallErrorMessage = useRef<string>("");
```

Comment:
This is being access from inside of an interval so need to use a ref to avoid stale state. We need to refactor it so that it fires a chain of promises instead of using an interval to inspect a value set by an async function.